src/global/logger.ts

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable no-console */
+import { config } from './config';
+import { LogLevelIndex, LogLevel } from './logger.type';
+
+interface LogMessage {
+  level: LogLevel;
+  message: string;
+  timestamp: string;
+  data?: any;
+}
+/**
+ * Logging class providing different levels of log
+ */
+class Logger {
```

Comment:
Probably makes sense to `export` so it is exposed to the documentation generator for the `API` section.

Reply:
I think I'll just add manual docs because it could confuse some users in this case.